### PR TITLE
Fix rootless cni netns cleanup logic and rename to rootless netns

### PR DESCRIPTION
**Pull request by Luap99**
#### What this PR does / why we need it:
Fix rootless cni netns cleanup logic

The check if cleanup is needed reads all container and checks if there
are running containers with bridge networking. If we do not find any we
have to cleanup the ns. However there was a problem with this because
the state is empty by default so the running check never worked.
Fortunately the was a second check which relies on the CNI files so we
still did cleanup anyway.

With netavark I noticed that this check is broken because the CNI files
were not present.

mount full XDG_RUNTIME_DIR in rootless cni ns

We should mount the full runtime directory into the namespace instead of
just the netns dir. This allows more use cases.

rename rootless cni ns to rootless netns

Since we want to use the rootless cni ns also for netavark we should
pick a more generic name. The name is now "rootless network namespace"
or short "rootless netns".

The rename might cause some issues after the update but when the
all containers are restarted or the host is rebooted it should work
correctly.

### CODE REVIEW

**Comment by Luap99** [comment]
@mheon PTAL

**Comment by mheon** [comment]
LGTM

File: cmd/podman/system/unshare.go [hunk 1]
```diff
@@ -34,7 +35,14 @@ func init() {
 	})
 	flags := unshareCommand.Flags()
 	flags.SetInterspersed(false)
-	flags.BoolVar(&unshareOptions.RootlessCNI, "rootless-cni", false, "Join the rootless network namespace used for CNI networking")
+	flags.BoolVar(&unshareOptions.RootlessNetNS, "rootless-netns", false, "Join the rootless network namespace used for CNI and netavark networking")
```
Comment (rhatdan) [inline cmd/podman/system/unshare.go:38]: Can you alias the old name.

**Pull request merged**